### PR TITLE
Buffer locking cleanup

### PR DESCRIPTION
**Pull request by K0bin**
Feels good to remove hacks.

Thanks to Axel Davy of Gallium Nine fame for telling me about the SWVP edge case here.

8bf11fd helps Injustice: Gods Among Us.

### CODE REVIEW

**Comment by K0bin** [comment]
I decided to keep the "allowDirectBufferMapping" option after all and enable it for Injustice.

**Comment by WinterSnowfall** [comment]
Does this also fix https://github.com/doitsujin/dxvk/issues/2375 mayhaps?

**Comment by K0bin** [comment]
@WinterSnowfall It doesn't, see my comment in the issue.

File: src/d3d9/d3d9_device.cpp [hunk 1]
```diff
@@ -4808,6 +4808,10 @@ namespace dxvk {
     if (desc.Usage & D3DUSAGE_DYNAMIC)
       Flags &= ~D3DLOCK_DONOTWAIT;
 
+    // Tests show that D3D9 drivers ignore DISCARD when the device is lost.
+    if (unlikely(m_deviceLostState != D3D9DeviceLostState::Ok))
```
Comment (Joshua-Ashton) [inline src/d3d9/d3d9_device.cpp:4809]: What about for images?

**Review by Joshua-Ashton — COMMENTED** [review]
I am sus about this commit but I guess we can see how it plays out...

**Pull request merged**